### DISCUSSION TITLE
How can we include complex terms in constructing the Hermitian Hamiltonian?

### DISCUSSION THREAD

**Discussion by thefacemyer** [QuSpin/QuSpin#579] [Q&A] [answered]
I am getting an error when I add a complex term in an XXZ Hamiltonian, specifically the z-component of the Dzyaloshinkii-Moriya interaction (i.e. J1+iDz). Is there a particular way of handling this sort of thing?

**Comment by JiahaoYao** [comment, marked as answer]
Would you like to show your code and your error message? 

Based on my previous experience, my guess is that you might need to refer to [this](https://github.com/weinbe58/QuSpin/discussions/527)

**Reply by thefacemyer** [reply]
Yes that reference fixed the problem. The hamiltonian was typed as float and not complex. My case was adding a nearest neighbor DMI z-component, and not an exponential term. Many thanks again!
